src/toolchain/toolchain.ts

```diff
@@ -181,12 +182,84 @@ export class SwiftToolchain {
      * @returns Folders for each Xcode install
      */
     public static async getXcodeInstalls(): Promise<string[]> {
+        if (process.platform !== "darwin") {
+            return [];
+        }
         const { stdout: xcodes } = await execFile("mdfind", [
             `kMDItemCFBundleIdentifier == 'com.apple.dt.Xcode'`,
         ]);
         return xcodes.trimEnd().split("\n");
     }
 
+    /**
+     * Reads the swiftly configuration file to find a list of installed toolchains.
+     *
+     * @returns an array of toolchain paths
+     */
+    public static async getSwiftlyToolchainInstalls(): Promise<string[]> {
+        try {
+            const swiftlyHomeDir: string | undefined = process.env["SWIFTLY_HOME_DIR"];
+            if (!swiftlyHomeDir) {
+                return [];
+            }
+            const swiftlyConfigRaw = await fs.readFile(
+                path.join(swiftlyHomeDir, "config.json"),
+                "utf-8"
+            );
+            const swiftlyConfig: unknown = JSON.parse(swiftlyConfigRaw);
+            if (
+                swiftlyConfig === undefined ||
+                swiftlyConfig === null ||
+                typeof swiftlyConfig !== "object" ||
+                !("installedToolchains" in swiftlyConfig)
+            ) {
+                return [];
+            }
+            const installedToolchains = swiftlyConfig.installedToolchains;
+            if (!Array.isArray(installedToolchains)) {
+                return [];
+            }
+            return installedToolchains
+                .filter((toolchain): toolchain is string => typeof toolchain === "string")
+                .map(toolchain => path.join(swiftlyHomeDir, "toolchains", toolchain));
+        } catch (error) {
+            throw new Error("Failed to retrieve Swiftly installations from disk.");
+        }
+    }
+
+    /**
+     * Checks common directories for available swift toolchain installations.
+     *
+     * @returns an array of toolchain paths
+     */
+    public static getToolchainInstalls(): Promise<string[]> {
```

Comment:
Is this Darwin only?

Reply:
Yup. The other platforms don't have common directories where toolchains get installed. I've added a short circuit at the beginning of this function to avoid checking on non-darwin platforms.